src/Logger.hpp

```diff
@@ -0,0 +1,325 @@
+// Logger class - for central logging to console/syslog/file
+//
+// Author  : Klaus-Dieter Gundermann
+// Created : 24.06.2020
+//
+// For all support, instructions and copyright go to:
+// http://e2guardian.org/
+// Released under the GPL v2, with the OpenSSL exception described in the README file.
+
+#ifndef __HPP_LOGGING
+#define __HPP_LOGGING
+
+#include <vector>
+//#include <sstream>
+#include <fstream>
+#include <sstream>
+#include <cstdio>
+#include <atomic>
+#include <syslog.h>
+#include <unistd.h>
+#include <sys/stat.h>
+
+// only C++14 : using namespace std::string_literals;
+
+class FileRec {
+public:
+    std::string filename;
+    FILE *file_stream;
+    int link_count = 0;
+    bool open = false;
+
+    bool write(std::string &msg);
+    bool rotate();
+    bool flush();
+};
+
+
+enum class LoggerSource {
+    none,
+    // used in production:
+            info, error, warning, accesslog, requestlog, storytrace, dstatslog,
+    // only usable when compiled with DEBUG_LOW:
+            debug, trace, network, story, regexp, config, content,
+    // only usable when compiled with DEBUG_HIGH:
+            icap, avscan, auth, dwload, proxy, thttps,
+    __Max_Value
+};
+
+enum class LoggerDestination {
+  none, stdout, stderr, syslog, file,
+  __Max_Value
+};
+
+class Logger {
+public:
+    Logger();   // constructor
+    ~Logger();  // destructor
+
+    void setSyslogName(const std::string logname);
```

Comment:
Codacy found an issue: [Function parameter 'logname' should be passed by const reference.](https://app.codacy.com/manual/numsys/e2guardian/pullRequest?prid=6238852)